orchestra/tests/test_query_params_filter.py

```diff
@@ -0,0 +1,95 @@
+from rest_framework.test import APIRequestFactory
+from rest_framework.viewsets import ModelViewSet
+
+from orchestra.models import Todo
+from orchestra.models import TodoQA
+from orchestra.tests.helpers.fixtures import TodoFactory
+from orchestra.tests.helpers.fixtures import ProjectFactory
+from orchestra.orchestra_api import build_url_params
+from orchestra.todos.filters import QueryParamsFilterBackend
+from orchestra.todos.serializers import TodoQASerializer
+from orchestra.tests.helpers import OrchestraTransactionTestCase
+from orchestra.todos.views import GenericTodoViewset
+
+
+factory = APIRequestFactory()
+
+
+class DummyView(ModelViewSet):
+    model = TodoQA
+    serializer_class = TodoQASerializer
+    queryset = TodoQA.objects.all()
+
+
+class QueryParamsFilterBackendTests(OrchestraTransactionTestCase):
+    def _get_qs_kwargs(self, view_class, url_params):
+        url = '/some-path/{}'.format(url_params)
+        view = view_class(action_map={'get': 'list'})
+        request = factory.get(url)
+        request = view.initialize_request(request)
+        view.request = request
+        view.format_kwarg = {}
+        backend = QueryParamsFilterBackend()
+        params = backend._get_params(request, view)
+        qs_kwargs = backend._get_filter_kwargs(view, params)
+        filterset_fields_kwargs = backend._get_filterset_fields_kwargs(
+            view, params, qs_kwargs)
+        return qs_kwargs, filterset_fields_kwargs
+
+    def test_random_key_values(self):
+        url_params = build_url_params(
+            123, 'slug', **{'some_random_key': 'some_value'})
+        qs_kwargs, _ = self._get_qs_kwargs(DummyView, url_params)
+        self.assertEqual(qs_kwargs, {})
+
+    def test_only_serializer_fields_are_passed(self):
+        """
+        In GenericTodoViewset we have the following filterset_fields:
+        filterset_fields = ('project__id', 'step__slug',)
+        A field which is not specified in it or in the view's
+        serializer fields cannot be passed as a filter args
+        """
+        title = 'Some title'
+        project_id = 123
+        url_params = build_url_params(
+            project_id,
+            None,
+            **{
+                'title': title,
+                'nonexistent_field': True})
+        qs_kwargs, filterset_fields_kwargs = self._get_qs_kwargs(
```

Comment:
I know that `_get_qs_kwargs ` uses the same logic as `filter_queryset` in `QueryParamsFilterBackend `. But I feel like if for some reason we change the logic in `filter_queryset`, the tests would not catch it. How about we create `_get_qs_kwargs`  in `QueryParamsFilterBackend` and test it directly. What do you think?

Reply:
I like your suggestion. Fixed :)